### DISCUSSION TITLE
Automatically disabling TS for the current buffer on an error.

### DISCUSSION THREAD

**Discussion by LordMZTE** [nvim-treesitter/nvim-treesitter#6447] [Q&A] [answered]
When nvim-treesitter inevitably breaks, perhaps due to out-of-date parsers/queries, I'd like to have it automatically disable TS for the current buffer (= call `vim.treesitter.stop()`) rather than assulting me with a virtually indestructible wall of errors. Is this possible?

**Comment by clason** [comment]
No, because the errors are caused during the decoration provider "hot loop"; there's nothing that can be done from this plugin. Things should be much better on Neovim 0.10, though.

**Reply by LordMZTE** [reply]
Hmm, I am on Neovim 0.10 and this still doesn't seem to be the case.

Note that I'm not even discussing the root cause of the issue here, I'm mostly looking for a workaround/quality of life feature.

**Reply by clason** [reply, marked as answer]
Yeah, but there is no workaround. We fix the root issue or go home. (Remember, treesitter is still experimental. That is no coincidence.)

**Reply by clason** [reply]
And things _have_ changed. If you don't see a difference, you're not on the latest master.
